services/supervisor/src/lib.rs

```diff
@@ -94,70 +138,63 @@ impl Service for Supervisor {
 
                 // Apply pending config in case 2/3+1 validators voted for it.
                 if confirmations >= byzantine_quorum(validators) {
-                    info!(
+                    log::info!(
                         "New configuration has been accepted: {:?}",
                         entry.config_propose
                     );
                     // Perform the application of configs.
-                    context.update_config(entry.config_propose.changes);
+                    update_configs(&mut context, entry.config_propose.changes);
+
                     // Remove config from proposals.
+                    let schema = Schema::new(context.instance().name, context.fork());
                     schema.pending_proposal().remove();
```

Comment:
Yes, it is. This is checked in `test_service_config_discard_single_apply_error` / `test_service_config_discard_single_apply_panic` in `tests/config.rs`. If you comment out this line, these tests fail (well, actually they don't right now since they use too permissive blockchain height, but it's trivial to fix).

Reply:
Hm. If I understand them correctly, they're checking applying the *service* config. And I'm talking about [`assert` in `update_configs`](https://github.com/exonum/exonum/pull/1509/files#diff-7a5cd928f4e93e0b1eceb23242deba18R74), which corresponds to the *consensus* config update.